src/main/java/nl/knaw/dans/ttv/core/TransferTask.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2021 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.ttv.core;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.dropwizard.hibernate.UnitOfWork;
+import nl.knaw.dans.ttv.db.TransferItemDAO;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.util.Objects;
+import java.util.zip.ZipFile;
+
+public class TransferTask extends Task {
+
+    private static final Logger log = LoggerFactory.getLogger(TransferTask.class);
+
+    public TransferTask(TransferItem transferItem, TransferItemDAO transferItemDAO) {
+        super(transferItem, transferItemDAO);
+    }
+
+    @Override
+    public TransferItem call() throws Exception {
+        log.info("Running task" + this);
+        extractMetadata();
+        return transferItem;
+    }
+
+    @UnitOfWork
+    public void extractMetadata() throws IOException {
+        ZipFile datasetVersionExport = new ZipFile(Paths.get(transferItem.getDveFilePath()).toFile());
+        String metadataFilePath = Objects.requireNonNull(datasetVersionExport.stream()
+                .filter(zipEntry -> zipEntry.getName().endsWith(".jsonld"))
+                .findAny().orElse(null)).getName();
```

Comment:
So, if the `.jsonld` file is not found a NPE is thrown? Can you make sure that something more descriptive is logged?